### PR TITLE
ACS-4027: Introduce changes to support removing tag from all associated nodes

### PR DESCRIPTION
**Pull request by gevagel**

### CODE REVIEW

File: remote-api/src/main/java/org/alfresco/rest/api/impl/TagsImpl.java [hunk 1]
```diff
@@ -128,37 +138,49 @@ public void deleteTag(String nodeId, String tagId)
...
-        {
-            Tag selectedTag = new Tag(pair.getFirst(), pair.getSecond());
-            selectedTag.setCount(tagsByCountMap.get(selectedTag.getTag()));
-            tags.add(selectedTag);
-        }
-
-        return CollectionWithPagingInfo.asPaged(paging, tags, results.hasMoreItems(), (totalItems == null ? null : totalItems.intValue()));
-    }
+    @Override
+	public void deleteTagById(StoreRef storeRef, String tagId) {
```
Comment (tpage-alfresco) [inline remote-api/src/main/java/org/alfresco/rest/api/impl/TagsImpl.java:142]: Tabs

Comment (gevagel) [reply remote-api/src/main/java/org/alfresco/rest/api/impl/TagsImpl.java:142]: Thanks for pointing that out Tom. Just to make sure, you are referring to this game with an acronym, right? [https://en.wikipedia.org/wiki/Totally_Accurate_Battle_Simulator](https://en.wikipedia.org/wiki/Totally_Accurate_Battle_Simulator) 😅

Comment (tpage-alfresco) [reply remote-api/src/main/java/org/alfresco/rest/api/impl/TagsImpl.java:142]: Ah yes of course!

**Review by tpage-alfresco — APPROVED** [review]
Other than the whitespace this looks good to me.

File: remote-api/src/test/java/org/alfresco/rest/api/impl/TagsImplTest.java [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import static org.junit.Assert.assertThrows;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.BDDMockito.then;
+
+@RunWith(MockitoJUnitRunner.class)
+public class TagsImplTest
```
Comment (tpage-alfresco) [inline remote-api/src/test/java/org/alfresco/rest/api/impl/TagsImplTest.java:48]: Hmm - why doesn't this new test class need to be added to a test suite? There's a test OmittedTestClassFinderUnitTest which I would have expected to pick this up.

**Review by krdabrowski — APPROVED** [review]

**Comment by gevagel** [comment]
After discussing with @krdabrowski, I will merge this to allow for other work to proceed, will follow up with a separate PR for E2E tests

**Pull request merged**